### ISSUE TITLE
Why is size of tensorrt compiled INT8 model after QAT is same as size of FP16 model

### ISSUE THREAD

**Issue by SM1991CODES** [pytorch/TensorRT#1379]
## ❓ Question

I have been trying to use INT8 inference for a trained pytorch model.

I followed this:
https://pytorch.org/TensorRT/_notebooks/vgg-qat.html
and
https://docs.nvidia.com/deeplearning/tensorrt/pytorch-quantization-toolkit/docs/tutorials/quant_resnet50.html

**My steps are outlined as:**
1) **Train pytorch model**

- Normal training loop

2) **Calibrate the model with quant_modules**

`
quant_modules.initialize(float_module_list=['ConvTranspose2d'])  # error when used in quantization, so I add it to the exception list
quant_desc_input = QuantDescriptor(calib_method='histogram')

# conv and Linear layers to be replaced by there quantized versions
quant_nn.QuantConv2d.set_default_quant_desc_input(quant_desc_input)
quant_nn.QuantLinear.set_default_quant_desc_input(quant_desc_input)
# quant_nn.QuantConvTranspose2d.set_default_quant_desc_input(quant_desc_input)

# now load pre-trained model
net1 = model_simple.BevDetNetSimple(input_channel_numbers, settings.N_CHANNELS_PREDICTION_KP, 
                                        scale_H=2, scale_W=2, predict_3d_center=True).cuda(DEVICE_ID_GPU)
cuda_dev = "cuda:{0}".format(DEVICE_ID_GPU)
net1.load_state_dict(torch.load(model_weights, map_location=cuda_dev))
print(net1)

calib_data = dataset_classes.CalibDataset(path_calib_dataset_bev=settings.val_bev_save_path)
calib_loader = DataLoader(calib_data, batch_size=6, drop_last=True)

# calibrate
with torch.no_grad():
    collect_stats(net1, calib_loader, num_batches=4)
    compute_amax(net1, method="percentile", percentile=99.99)
`

3. **Train the model with quantized layers**

- Normal training loop, 50 epochs
- Save as torchscript

`# export to torchscript
quant_nn.TensorQuantizer.use_fb_fake_quant = True
with torch.no_grad():
    jit_model = torch.jit.trace(net1, train_x)
    torch.jit.save(jit_model, settings.MODEL_SAVE_PATH_INTERIM + str(epoch) + '_qat.ts')
`

4. **Convert QAT trained torchscript to tensorrt - int8**

`
def export_qat_to_trt_int8(path_trained_qat_ts, path_save_ts_trt_int8):
    """
    Function exports the QAT trained model saved as torchscript and weighst to tensorrt usig INT8 precision
    """

    # load the saved QAT ts model
    qat_model = torch.jit.load(path_trained_qat_ts).eval()
    
    # compile to torchscript
    compile_spec = {"inputs": [trt.Input([1, 4, 384, 384])],
                "enabled_precisions": [torch.int8], 
                "truncate_long_and_double":True,
                "sparse_weights": True}
    trt_mod = trt.compile(qat_model, **compile_spec)

    torch.jit.save(trt_mod, path_save_ts_trt_int8)
`

After doing the above steps, I get a model of size 48MB, which is same as that of FP16 model. The runtime is also similar.

I have then tried PTQ technique for INT8 - this gives me a model of size 28MB, which is expected. Also, the runtime is about half of FP16 model. This is fine. However, the accuracy is not acceptable.

Please let me know what am I missing in the QAT way? Why is my model larger and slower wrt PTQ?

## Environment

> Build information about Torch-TensorRT can be found by turning on debug messages

 - PyTorch Version (e.g., 1.0): 1.12
 - CPU Architecture: x86_64
 - OS (e.g., Linux): linux
 - How you installed PyTorch (`conda`, `pip`, `libtorch`, source): pip
 - Build command you used (if compiling from source):
 - Are you using local sources or building from archives:
 - Python version: 3.8.10
 - CUDA version: 11.6
 - GPU models and configuration: RTX3090/ RTX2080 MAXQ
 - Any other relevant information:

**Comment by godhj93** [comment]
I have same problem. did you solve it?
